rust/core-lib/src/editor.rs

```diff
@@ -357,8 +359,174 @@ impl Editor {
                     && config.use_tab_stops {
                     tab_start
                 } else {
-                    self.text.prev_grapheme_offset(region.end)
-                        .unwrap_or(region.end)
+
+                    #[derive(PartialEq)]
+                    enum State {
+                        Start,
+                        LF,
```

Comment:
Nit: Rust idiom is to only capitalize the first letter of acronyms/initialisms, so this would be `Lf`, and below would be `BeforeZwj`, etc.